PyPlaque/utils/fp_readout_object.py

```diff
@@ -78,23 +79,58 @@ def get_major_minor_axis_length(self):
         return self.plaque_object_properties.axis_major_length, \
                 self.plaque_object_properties.axis_minor_length
     
+    def get_custom_eccentricity(self,):
+        # Calculate the image moments
+        y_indices, x_indices = np.nonzero(binary_image)
+        x_mean = x_indices.mean()
+        y_mean = y_indices.mean()
+        x_indices_centered = x_indices - x_mean
+        y_indices_centered = y_indices - y_mean
+
+        # Calculate central moments
+        mu11 = (x_indices_centered * y_indices_centered).sum()
+        mu20 = (x_indices_centered * x_indices_centered).sum()
+        mu02 = (y_indices_centered * y_indices_centered).sum()
+
+        # Calculate the axis lengths of the ellipse (major_axis_length, minor_axis_length)
+        # The math comes from the eigenvalues of the covariance matrix of the object's x and y pixel coordinates
+        a = mu20 + mu02
+        b = np.sqrt((mu20 - mu02) ** 2 + 4 * mu11 ** 2)
+        major_axis_length = np.sqrt(2) * np.sqrt(a + b)
+        minor_axis_length = np.sqrt(2) * np.sqrt(a - b)
+
+        # Calculate eccentricity
+        eccentricity = np.sqrt(1 - (minor_axis_length / major_axis_length) ** 2)
+
+        return eccentricity
+
+
     def get_eccentricity(self):
+        
         return self.plaque_object_properties.eccentricity
     
     def get_convex_area(self):
         return self.plaque_object_properties.area_convex
     
     def roundness(self):
-        point1 = np.array((self.plaque_object_properties.bbox[3],
-                           self.plaque_object_properties.bbox[2]))
-        point2 = np.array(((self.plaque_object_properties.bbox[3] \
-                            +self.plaque_object_properties.bbox[1])/2,
-                            (self.plaque_object_properties.bbox[2] \
-                             +self.plaque_object_properties.bbox[0])/2))
-        radius = np.linalg.norm(point1 - point2)
-        perimeter = 2 * np.pi * radius
-        ratio = 4 * np.pi * self.get_area() / ( perimeter ** 2 ) 
-        return 1-ratio
+
+
+
+        image = self.plaque_object_properties.image_convex 
+        area = self.plaque_object_properties.area_convex
+        #  Find contours
+        contours = skimage.measure.find_contours(image)
+
+        # Assuming the largest contour corresponds to the object
+        contour = max(contours, key=len)
+
+        # Calculate the perimeter
+        perimeter = np.sum(np.sqrt(np.sum(np.diff(contour, axis=0)**2, axis=1)))
+        perimeter = self.plaque_object_properties.perimeter
+        
+        roundness = (4 * np.pi * area) / ( (perimeter ** 2))
+        
+        return perimeter,area
```

Comment:
Why do we return perimeter, area instead of the roundness value we calculated here ?